cmd/add-member.cmd.ts

```diff
@@ -0,0 +1,125 @@
+import fs from "fs"
+import fetch from "node-fetch"
+import fileType from "file-type"
+import inquirer from "inquirer"
+import { Command } from "./index";
+
+const IMAGE_DIR = "./public/avatars";
+const MEMBERS_DIR = "./data/members";
+
+type Answer = {
+  username: string;
+  title: string;
+  designation: string;
+  company: string;
+  imageUrl: string;
+};
+
+async function downloadAndSaveImg(answer: Answer) {
+  try {
+    const response = await fetch(answer.imageUrl);
+    const buffer = await response.buffer();
+    const fileDetails = await fileType.fromBuffer(buffer);
+    const fileName = `${answer.username}.${fileDetails.ext}`;
+
+    fs.writeFile(`${IMAGE_DIR}/${fileName}`, buffer, () => {
+      console.log("Image saved!");
+    });
+
+    return `/avatars/${fileName}`;
+  } catch (e) {
+    console.log("Something went wrong, couldn't save image;", e);
+  }
+}
+
+function validateMinLength(input: string) {
+  if (input.length < 4) {
+    return "length must be at least 4";
+  } else {
+    return true;
+  }
+}
+
+async function validateUsername(username: Answer["username"]) {
+  if (username.length < 4) {
+    return "length must be at least 4";
+  } else {
+    const directory = `${MEMBERS_DIR}/${username}`;
+
+    if (!fs.existsSync(directory)) {
+      fs.mkdirSync(directory);
+      return true;
+    } else {
+      return "username already exist";
+    }
+  }
+}
+
+async function addMember() {
```

Comment:
No need for this separate function, let's put the logic inside the same command body